pkg/api/controller_test.go

```diff
@@ -200,32 +200,11 @@ func TestRunAlreadyRunningServer(t *testing.T) {
 		conf := config.New()
 		conf.HTTP.Port = port
 
-		ctlr := api.NewController(conf)
-
-		globalDir := t.TempDir()
-
-		ctlr.Config.Storage.RootDirectory = globalDir
-
-		go func() {
-			if err := ctlr.Run(context.Background()); err != nil {
-				return
-			}
-		}()
-
-		// wait till ready
-		for {
-			_, err := resty.R().Get(baseURL)
-			if err == nil {
-				break
-			}
-
-			time.Sleep(100 * time.Millisecond)
-		}
+		ctlr := makeController(conf, t.TempDir(), "")
 
-		defer func() {
-			ctx := context.Background()
-			_ = ctlr.Server.Shutdown(ctx)
-		}()
+		go startServer(ctlr)
```

Comment:
```
ctlr := test.NewController(...)
go ctlr.Start(wait=true/false)
defer ctlr.Stop()
```

Reply:
Made these changes, but now I see the part where we specify the wait=true/false. Will also add this

Reply:
Now when wait would be true, we call go ctlr.Start(baseUrl), or ctlr.Start("") when wait would be false, since url is needed

Reply:
ctlr.Start(wait=true/false)?